streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedStreamAggregateBuilder.java

```diff
@@ -31,90 +26,90 @@
+                                                       final Initializer<VOut> initializer,
+                                                       final NamedInternal named,
+                                                       final StoreBuilder<? extends StateStore> storeBuilder,
+                                                       final Serde<KR> keySerde,
+                                                       final Serde<VOut> valSerde,
+                                                       final String queryableName,
+                                                       final Windows<W> windows,
+                                                       final SessionWindows sessionWindows,
+                                                       final Merger<? super K, VOut> sessionMerger) {
 
         final Collection<? extends AbstractStream<K, ?>> groupedStreams = new ArrayList<>(groupPatterns.keySet());
         final AbstractStream<K, ?> kGrouped = groupedStreams.iterator().next();
         groupedStreams.remove(kGrouped);
         kGrouped.ensureCopartitionWith(groupedStreams);
 
-
         final Collection<StreamsGraphNode> processors = new ArrayList<>();
         boolean stateCreated = false;
         int counter = 0;
-        for (final Entry<KGroupedStreamImpl<K, ?>, Aggregator<? super K, ? super Object, VOut>> kGroupedStream : groupPatterns
-                .entrySet()) {
+        for (final Entry<KGroupedStreamImpl<K, ?>, Aggregator<? super K, ? super Object, VOut>> kGroupedStream : groupPatterns.entrySet()) {
             final StatefulProcessorNode statefulProcessorNode = getStatefulProcessorNode(
-                    kGroupedStream.getValue(),
-                    initializer,
-                    named.withName(named.suffixWithOrElseGet("-cogroup-agg-" + counter++, builder, CogroupedKStreamImpl.AGGREGATE_NAME)),
-                    stateCreated,
-                    storeBuilder,
-                    windows,
-                    sessionWindows,
-                    sessionMerger);
+                kGroupedStream.getValue(),
+                initializer,
+                named.suffixWithOrElseGet(
+                    "-cogroup-agg-" + counter++,
+                    builder,
+                    CogroupedKStreamImpl.AGGREGATE_NAME),
+                stateCreated,
+                storeBuilder,
+                windows,
+                sessionWindows,
+                sessionMerger);
             stateCreated = true;
             processors.add(statefulProcessorNode);
             builder.addGraphNode(kGroupedStream.getKey().streamsGraphNode, statefulProcessorNode);
         }
-        final String mergeProcessorName = named.orElseGenerateWithPrefix(builder, CogroupedKStreamImpl.AGGREGATE_NAME);
+        final String mergeProcessorName = named.suffixWithOrElseGet(
+            "-cogroup-merge",
```

Comment:
Changed this to generate a name `<userName>-cogroup-merge` to align to `<userName>-cogroup-agg-<counter>` instead of just `<userName>` for the merge node.